✨[Feat] : 예외 처리

Yejin-Moon opened the pull request:
예외 처리 추가

### Pull Request 타입(하나 이상의 PR 타입을 선택해주세요)
- [x] 기능 추가
- [ ] 기능 삭제
- [ ] 버그 수정
- [ ] 의존성, 환경 변수, 빌드 관련 코드 업데이트

### 반영 브랜치
SPOFO-174 -> main

### 변경 사항
예외 처리 기능 추가

### 테스트 결과
1) TOKEN_EXPIRED → 401
2) INVALID_TOKEN → 401
3) ID_NOT_EXIST → 401
4) SERVER_ERROR → 500
위의 에러를 반환하도록 함

Yejin-Moon committed:
✨[Feat] : 예외 처리

예외 처리 추가

Yejin-Moon committed:
Merge branch 'main' into SPOFO-174

JANG-WON-JUN commented on src/main/java/net/spofo/auth/exception/ErrorCode.java:
```diff
@@ -0,0 +1,24 @@
+
+    @Override
+    public ErrorResult getErrorResult() {
+        return ErrorResult.builder()
```
빌더를 사용할 떄는 필드 별로 개행을 해서 작성하는게 좋을 것 같아요.

JANG-WON-JUN commented on src/main/java/net/spofo/auth/exception/advice/ExceptionControllerAdvice.java:
```diff
@@ -0,0 +1,29 @@
+                .code(errorCode)
+                .reason(errorMsg)
+                .build();
+        return new ResponseEntity<>(errorResult, HttpStatus.valueOf(status));
```
ResponseEntity의 static method를 사용해서 반환하면 좋을 것 같아요

JANG-WON-JUN commented on src/main/java/net/spofo/auth/exception/advice/ExceptionControllerAdvice.java:
```diff
@@ -0,0 +1,29 @@
+@RequiredArgsConstructor
+public class ExceptionControllerAdvice {
+
+    @ResponseBody
```
RestControllerAdvice가 있으면 ResponseBody가 필요없을 것 같아요

Yejin-Moon committed:
🎨[Style] : 코드 수정

1. ResponseEntity의 static method를 사용해서 반환하도록 변경
2. error 발생 시 생성하도록 수정
3. 빌더 개행 수정

Yejin-Moon committed:
✨[Feat] : 서명 검증 기능 추가

1. 공개키를 이용한 서명 검증 기능 추가
2. 예외처리 기능 수정

Yejin-Moon committed:
✨[Feat] : n, e값 추가

1. 공개키를 파싱해 올 때 n, e도 DB에 저장
2. DB에 저장된 n, e를 이용해서 공개키 생성해서 서명 검증

keg51051 commented on src/main/java/net/spofo/auth/entity/PublicKey.java:
```diff
@@ -16,17 +16,25 @@
+    public PublicKey(String publickey, String n, String e) {
         this.publickey = publickey;
+        this.n=n;
+        this.e=e;
```
n과 e가 나타내는게 뭔가요?

Yejin-Moon replied:
n과 e는 RSA 공개키를 구성하는 값입니다. n은 modulus, e는 exponent로 변경하겠습니다.

Yejin-Moon committed:
♻[Rename] : 메서드 이름 변경

Yejin-Moon committed:
♻[Rename] : 변수명 변경(n, e)

n -> modulus
e -> exponent

keg51051 reviewed (approved):
lgtm

JANG-WON-JUN commented on src/main/java/net/spofo/auth/entity/PublicKey.java:
```diff
@@ -16,17 +16,25 @@
     private String publickey;
 
+    @Column(length = 500, nullable = false)
+    private String modulus;
```
컬럼별로 줄바꿈을 통일 해주시면 좋을 것 같습니다.

JANG-WON-JUN commented on src/main/java/net/spofo/auth/entity/PublicKey.java:
```diff
@@ -16,17 +16,25 @@
+    private String exponent;
+
+
+
```
줄바꿈 확인 부탁드립니다.

JANG-WON-JUN commented on src/main/java/net/spofo/auth/entity/PublicKey.java:
```diff
@@ -16,17 +16,25 @@
+
     @Builder
-    public PublicKey(String publickey) {
+    public PublicKey(String publickey, String modulus, String exponent) {
```
다른 곳의 builder를 붙인 생성자는 접근 제한자가 private인데, 여기는 public으로 되어있어서 확인 부탁드립니다.

Yejin-Moon committed:
🐛[Fix] : 리턴값 변경

verfySignature 에서 리턴 한 값을 사용하지 않으므로 void를 리턴하도록 변경

Yejin-Moon committed:
🎨[Style] : 줄바꿈/private 수정
